Suggestion to import item via `pub use X as _` re-export, which doesn't work

### Code

```Rust
#![allow(unused_imports)]

mod inner {
    pub enum Example {
        ExOne,
        ExTwo,
    }
}

mod reexports {
    pub use crate::inner::{
        Example as _,
    };
}

use crate::reexports::*;

fn check() {
    ExOne;
}
```


### Current output

```Shell
error[E0425]: cannot find value `ExOne` in this scope
  --> src/lib.rs:19:5
   |
19 |     ExOne;
   |     ^^^^^ not found in this scope
   |
help: consider importing this unit variant
   |
16 + use crate::_::ExOne;
   |

For more information about this error, try `rustc --explain E0425`.
```


### Desired output

```Shell
error[E0425]: cannot find value `ExOne` in this scope
  --> src/lib.rs:19:5
   |
19 |     ExOne;
   |     ^^^^^ not found in this scope
   |
help: consider importing this unit variant
   |
16 + use crate::inner::ExOne;
   |

For more information about this error, try `rustc --explain E0425`.
```


### Rationale and extra context

The suggested fix doesn't work, since `_` is a reserved identifier and items may not be imported through it:
```Shell
error: expected identifier, found reserved identifier `_`
 --> src/lib.rs:1:12
  |
1 | use crate::_::ExOne;
  |            ^ expected identifier, found reserved identifier
``` 

The appropriate suggestion is to import the item via a valid path.

### Other cases

_No response_

### Anything else?

Playground: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=013d5f59846dad0ae3add3c12d24c755

Came up as a result of triaging: https://github.com/obi1kenobi/cargo-semver-checks/issues/536

--- a/compiler/rustc_resolve/src/diagnostics.rs
+++ b/compiler/rustc_resolve/src/diagnostics.rs
@@ -1169,6 +1169,10 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
                     return;
                 }
 
+                if ident.name == kw::Underscore {
+                    return;
+                }
+
                 let child_accessible =
                     accessible && this.is_accessible_from(name_binding.vis, parent_scope.module);
 

--- a/tests/ui/resolve/issue-116164.rs
+++ b/tests/ui/resolve/issue-116164.rs
@@ -1,0 +1,19 @@
+#![allow(unused_imports)]
+
+mod inner {
+    pub enum Example {
+        ExOne,
+    }
+}
+
+mod reexports {
+    pub use crate::inner::Example as _;
+}
+
+use crate::reexports::*;
+//~^ SUGGESTION: use inner::Example::ExOne
+
+fn main() {
+    ExOne;
+    //~^ ERROR: cannot find value `ExOne` in this scope
+}

--- a/tests/ui/resolve/issue-116164.stderr
+++ b/tests/ui/resolve/issue-116164.stderr
@@ -1,0 +1,14 @@
+error[E0425]: cannot find value `ExOne` in this scope
+  --> $DIR/issue-116164.rs:17:5
+   |
+LL |     ExOne;
+   |     ^^^^^ not found in this scope
+   |
+help: consider importing this unit variant
+   |
+LL + use inner::Example::ExOne;
+   |
+
+error: aborting due to previous error
+
+For more information about this error, try `rustc --explain E0425`.

